src/analysis/processing/qgsalgorithmtinmeshcreation.cpp

```diff
@@ -0,0 +1,180 @@
+      mAvailableFormat.append( driverMeta.name() );
+
+  addParameter( new QgsProcessingParameterEnum( QStringLiteral( "MESH_FORMAT" ), QObject::tr( "Output format" ), mAvailableFormat, false, 0 ) );
+  addParameter( new QgsProcessingParameterCrs( QStringLiteral( "CRS_OUTPUT" ), QObject::tr( "Output Coordinate System" ), QVariant(), true ) );
+  addParameter( new QgsProcessingParameterFileDestination( QStringLiteral( "OUTPUT_MESH" ), QObject::tr( "Output File" ) ) );
+}
+
+bool QgsTinMeshCreationAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  Q_UNUSED( feedback );
+  const QVariant layersVariant = parameters.value( parameterDefinition( QStringLiteral( "SOURCE_DATA" ) )->name() );
+  if ( layersVariant.type() != QVariant::List )
+    return false;
+
+  const QVariantList layersList = layersVariant.toList();
+
+  QgsCoordinateReferenceSystem destinationCrs = parameterAsCrs( parameters, QStringLiteral( "CRS_OUTPUT" ), context );
+  if ( !destinationCrs.isValid() )
+    destinationCrs = context.project()->crs();
+
+  for ( const QVariant &layer : layersList )
+  {
+    if ( feedback && feedback->isCanceled() )
+      return false;
+
+    if ( layer.type() != QVariant::Map )
+      continue;
+    const QVariantMap layerMap = layer.toMap();
+    const QString layerId = layerMap.value( QStringLiteral( "Id" ) ).toString();
+    const QgsProcessingParameterTinInputLayers::Type type =
+      static_cast<QgsProcessingParameterTinInputLayers::Type>( layerMap.value( QStringLiteral( "Type" ) ).toInt() );
+    int attributeIndex = layerMap.value( QStringLiteral( "AttributeIndex" ) ).toInt();
+
+    QgsVectorLayer *vectorLayer = context.project()->mapLayer<QgsVectorLayer *>( layerId );
+    if ( !vectorLayer || !vectorLayer->isValid() )
+      continue;
+
+    const QgsCoordinateTransform transform( vectorLayer->crs(), destinationCrs, context.transformContext() );
+    int featureCount = vectorLayer->featureCount();
+    switch ( type )
+    {
+      case QgsProcessingParameterTinInputLayers::Vertices:
+        mVerticesLayer.append( {vectorLayer->getFeatures(), transform, attributeIndex, featureCount} );
+        break;
+      case QgsProcessingParameterTinInputLayers::BreakLines:
+        mBreakLinesLayer.append( {vectorLayer->getFeatures(), transform, attributeIndex, featureCount} );
+        break;
+    }
+  }
+
+  if ( mVerticesLayer.isEmpty() && mBreakLinesLayer.isEmpty() )
+    return false;
+
+  return true;
+}
+
+QVariantMap QgsTinMeshCreationAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  QgsMeshTriangulation triangulation;
+  QgsCoordinateReferenceSystem destinationCrs = parameterAsCrs( parameters, QStringLiteral( "CRS_OUTPUT" ), context );
+  if ( !destinationCrs.isValid() )
+    destinationCrs = context.project()->crs();
```

Comment:
we won't always have project in the context, so you'll need to check if it's set before referencing here